fhir-model/src/main/java/com/ibm/fhir/model/util/JsonSupport.java

```diff
@@ -78,11 +80,39 @@ public static void init() { }
         }
         return Collections.unmodifiableMap(elementNameMap);
     }
+    
+    
+    private static Map<Class<?>, Set<String>> buildSummaryElementNameMap(boolean isData) {
```

Comment:
what is `isData`?  maybe some documentation to better describe that input param?

Reply:
I had the same question.

Reply:
_summary need to support values:
    true
    text
    data
    count

https://www.hl7.org/fhir/search.html#summary

Reply:
if summary="data" is just "all fields except `text`" then do we really need a separate map for it?

Reply:
should we use an enumeration for the different values?

Reply:
yes, we do. otherwise, we will have to generate the list every time, or we will have to change the current element filtering codes to skip "text" filed. 

yes, we can also use enumeration. it doesn't matter too much.

Reply:
I did considere using Enum, but it will make the codes a little bit complicated, but yes, I can change to use Enum for the values.

Reply:
en... after think it over again and tried to make some code changes to use Enum, I found seems it's not a good idea to use Enum, because we compares the String value, and also check if a input stream is among the string values of the Enum.  seems a little bit overkill here.   But I do think I should change to use Set instead of list for the values.

Reply:
the new `getSummaryDataElementNames` impl looks good to me